internal/services/policy/assignment_base_resource.go

```diff
@@ -326,6 +336,26 @@ func (br assignmentBaseResource) arguments(fields map[string]*pluginsdk.Schema)
 			},
 		},
 
+		"non_compliance_message": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"message": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+
+					"policy_definition_reference_id": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
and could we validate this with the policy reference id validator?

Reply:
Hi,

This isn't a resource ID, this is a string value to identity each policy definition within a set. For an example see here:

https://docs.microsoft.com/en-us/azure/governance/policy/concepts/assignment-structure#non-compliance-messages

And here:

https://docs.microsoft.com/en-us/azure/templates/microsoft.authorization/policysetdefinitions?tabs=bicep#policydefinitionreference

Where it's defined as:

policyDefinitionReferenceId - A unique id (within the policy set definition) for this policy definition reference. (string)

Reply:
Resolving for now as `string` is correct here